lnwallet/chancloser/chancloser.go

```diff
@@ -526,80 +594,73 @@ func (c *ChanCloser) ProcessCloseMsg(msg lnwire.Message) ([]lnwire.Message,
 				"have %v", spew.Sdump(msg))
 		}
 
-		// As we're the responder to this shutdown (the other party
-		// wants to close), we'll check if this is a frozen channel or
-		// not. If the channel is frozen and we were not also the
-		// initiator of the channel opening, then we'll deny their close
-		// attempt.
+		// Populate the Shutdown bool to catch if we receive two
+		// Shutdown messages from ourselves or the peer. Receiving a
+		// second from ourselves shouldn't be possible, but we catch it
+		// anyways.
+		if remote {
+			c.receivedRemoteShutdown = true
+		} else {
+			c.receivedLocalShutdown = true
```

Comment:
Hmm, yeh this flips the flow a bit: at this point before it's assumed that this is only executed when we _receive_ a shutdown, since we initiate it via another route.

Reply:
Generally I think the old logic was easier to follow, as we don't need the local/remote distinction in many of these state transitions as the state itself determines which "side" is processing.